From 780a0e551bec2035d1fa72636986d42d35584aba Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Wed, 23 Nov 2016 10:16:38 +0800
Subject: [PATCH] [MMFMWK-7403] Remove ion allocator and change the dependence
 to gst-based

ion allocator has been move to gst-base in [MMFMWK-7394]

Upstream Status: Inappropriate [i.MX specific]

Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 configure.ac                     | 11 +++++------
 ext/gl/gstgldownloadelement.c    |  2 +-
 gst-libs/gst/gl/Makefile.am      |  7 -------
 gst-libs/gst/gl/gstglmemorydma.c |  6 +++---
 gst-libs/gst/gl/gstglupload.c    |  8 ++++----
 5 files changed, 13 insertions(+), 21 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7e9041d..8416099 100644
--- a/configure.ac
+++ b/configure.ac
@@ -581,11 +581,6 @@ AG_GST_CHECK_FEATURE(ION, [ion], ion, [
   AC_CHECK_HEADER(linux/ion.h, HAVE_ION="yes", HAVE_ION="no")
 ])
 
-GST_GL_HAVE_ION=0
-if test "x$HAVE_ION" = "xyes"; then
-  GST_GL_HAVE_ION=1
-fi
-
 dnl *** opengl ***
 AC_ARG_ENABLE([opengl],
      [  --enable-opengl         Enable Desktop OpenGL support @<:@default=auto@:>@],
@@ -880,10 +875,14 @@ PKG_CHECK_MODULES(GST_ALLOCATORS, gstreamer-allocators-1.0,
   HAVE_GST_ALLOCATORS=yes, )
 
 GST_GL_HAVE_DMABUF=0
+GST_GL_HAVE_IONDMA=0
 if test "x$HAVE_DRM_FOURCC_HEADER" = "xyes" -a \
         "x$HAVE_GST_ALLOCATORS" = "xyes" -a \
         "x$HAVE_EGL" = "xyes"; then
           GST_GL_HAVE_DMABUF=1
+  if test "x$HAVE_ION" = "xyes"; then
+    GST_GL_HAVE_IONDMA=1
+  fi
 fi
 
 dnl check if we can include both GL and GLES2 at the same time
@@ -1292,7 +1291,7 @@ GL_CONFIG_DEFINES="$GL_CONFIG_DEFINES
 
 GL_CONFIG_DEFINES="$GL_CONFIG_DEFINES
 #define GST_GL_HAVE_DMABUF $GST_GL_HAVE_DMABUF
-#define GST_GL_HAVE_ION $GST_GL_HAVE_ION
+#define GST_GL_HAVE_IONDMA $GST_GL_HAVE_IONDMA
 #define GST_GL_HAVE_DIRECTVIV $GST_GL_HAVE_DIRECTVIV
 #define GST_GL_HAVE_PHYMEM $GST_GL_HAVE_PHYMEM
 "
diff --git a/ext/gl/gstgldownloadelement.c b/ext/gl/gstgldownloadelement.c
index d6ccc6e..a20da0d 100644
--- a/ext/gl/gstgldownloadelement.c
+++ b/ext/gl/gstgldownloadelement.c
@@ -29,7 +29,7 @@
 #include <gst/gl/gstglphymemory.h>
 #endif
 
-#if GST_GL_HAVE_DMABUF
+#if GST_GL_HAVE_DMABUF && GST_GL_HAVE_IONDMA
 #include <gst/gl/gstglmemorydma.h>
 #endif
 
diff --git a/gst-libs/gst/gl/Makefile.am b/gst-libs/gst/gl/Makefile.am
index ccca834..584f47f 100644
--- a/gst-libs/gst/gl/Makefile.am
+++ b/gst-libs/gst/gl/Makefile.am
@@ -105,8 +105,6 @@ libgstgl_@GST_API_VERSION@_la_LIBADD += -lg2d
 endif
 
 if USE_ION
-libgstgl_@GST_API_VERSION@_la_LIBADD += \
-	$(top_builddir)/gst-libs/gst/ion/libgstbadion-$(GST_API_VERSION).la
 if USE_EGL
 libgstgl_@GST_API_VERSION@_la_SOURCES += gstglmemorydma.c
 libgstgl_@GST_API_VERSION@include_HEADERS += gstglmemorydma.h
@@ -200,10 +198,6 @@ GstGL-@GST_API_VERSION@.gir: $(INTROSPECTION_SCANNER) libgstgl-@GST_API_VERSION@
 		$(GL_CFLAGS) \
 		--add-include-path=$(PKG_CONFIG_SYSROOT_DIR)`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-@GST_API_VERSION@` \
 		--add-include-path=$(PKG_CONFIG_SYSROOT_DIR)`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-base-@GST_API_VERSION@` \
-		--add-include-path="$(top_builddir)/gst-libs/gst/ion/" \
-		--library-path=$(PKG_CONFIG_SYSROOT_DIR)`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=libdir gstreamer-@GST_API_VERSION@` \
-		--library-path=$(PKG_CONFIG_SYSROOT_DIR)`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=libdir gstreamer-base-@GST_API_VERSION@` \
-		--library-path="$(top_builddir)/gst-libs/gst/ion/" \
 		--library=libgstgl-@GST_API_VERSION@.la \
 		--include=Gst-@GST_API_VERSION@ \
 		--include=GstBase-@GST_API_VERSION@ \
@@ -234,7 +228,6 @@ typelibs_DATA = $(BUILT_GIRSOURCES:.gir=.typelib)
 		--includedir=$(builddir) \
 		--includedir=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-@GST_API_VERSION@` \
 		--includedir=`PKG_CONFIG_PATH="$(GST_PKG_CONFIG_PATH)" $(PKG_CONFIG) --variable=girdir gstreamer-base-@GST_API_VERSION@` \
-		--includedir="$(top_builddir)/gst-libs/gst/ion/" \
 		$(INTROSPECTION_COMPILER_OPTS) $< -o $(@F)
 
 CLEANFILES = $(BUILT_GIRSOURCES) $(typelibs_DATA)
diff --git a/gst-libs/gst/gl/gstglmemorydma.c b/gst-libs/gst/gl/gstglmemorydma.c
index fa418a1..1589f65 100644
--- a/gst-libs/gst/gl/gstglmemorydma.c
+++ b/gst-libs/gst/gl/gstglmemorydma.c
@@ -30,8 +30,8 @@
 #include <gst/allocators/gstdmabuf.h>
 #include <gst/gl/gstglmemorydma.h>
 
-#if GST_GL_HAVE_ION
-#include <gst/ion/gstionmemory.h>
+#if GST_GL_HAVE_IONDMA
+#include <gst/allocators/gstionmemory.h>
 #endif
 
 GST_DEBUG_CATEGORY_STATIC (GST_CAT_GL_DMA_MEMORY);
@@ -49,7 +49,7 @@ gst_gl_memory_dma_init_instance (void)
 
   GST_DEBUG_CATEGORY_INIT (GST_CAT_GL_DMA_MEMORY, "glmemorydma", 0, "OpenGL dma memory");
 
-#if GST_GL_HAVE_ION
+#if GST_GL_HAVE_IONDMA
   ion_allocator = gst_ion_allocator_obtain();
 #endif
 
diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index fbef086..c1de1a8 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -36,8 +36,8 @@
 #include <gst/allocators/gstdmabuf.h>
 #endif
 
-#if GST_GL_HAVE_ION
-#include <gst/ion/gstionmemory.h>
+#if GST_GL_HAVE_IONDMA
+#include <gst/allocators/gstionmemory.h>
 #endif
 
 #if GST_GL_HAVE_DIRECTVIV
@@ -774,7 +774,7 @@ _dma_buf_upload_setup_buffer_pool (GstBufferPool **pool, GstAllocator *allocator
 
   /* if user not provide an allocator, then use default ion allocator*/
   if (!allocator) {
-#if GST_GL_HAVE_ION
+#if GST_GL_HAVE_IONDMA
     allocator = gst_ion_allocator_obtain ();
 #endif
   }
@@ -991,7 +991,7 @@ _dma_buf_upload_propose_allocation (gpointer impl, GstQuery * decide_query,
   if (!gst_video_info_from_caps (&info, caps))
     goto invalid_caps;
 
-#if GST_GL_HAVE_ION
+#if GST_GL_HAVE_IONDMA
   allocator = gst_ion_allocator_obtain ();
 #endif
   if (!allocator) {
-- 
2.9.0.GIT

